### PR TITLE
FS-2563: Send email for incomplete applications

### PR DESCRIPTION
**Pull request by ksp37-dluhc**
Ticket : https://dluhcdigital.atlassian.net/jira/software/c/projects/FS/boards/50?selectedIssue=FS-2563

The ticket above describes the requirement to send a copy of any incomplete applications. This is a type of notification, similar to the existing 'deadline reminder' notification that currently exists. We now have multiple notifications that need to be managed (and perhaps more in future). More generally, these can be considered as types of 'events' that need to be processed during fund rounds. This PR adds two functions, one to pull events from the fund store and hand them over to the relevant processor, and another (a processor) to send applicants a copy of their incomplete applications.  

Change description
A brief description of the pull request

[N/A] Unit tests and other appropriate tests added or updated
[N/A] README and other documentation has been updated / added (if needed)
[Done] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")
How to test
QA testing will require creation of an incomplete application for a fund round, and a dev to create an (active) event in the fund store. The lambda function should pull the event and send an email for the incomplete application. 

Screenshots of UI changes (if applicable)
N/A

### CODE REVIEW

File: apps/pre-award/lambdas/application-deadline-reminder/data.py [hunk 1]
```diff
@@ -27,7 +27,21 @@ def get_data(endpoint, params: Optional[dict] = None):
         logging.error("There was a problem retrieving response from"
         f" {endpoint}. Status code: {response.status_code}")
         return None
+    
+def get_data_safe(endpoint, params: Optional[dict] = None):
+    try:
+        response = requests.get(endpoint, params=params)
+        response.raise_for_status()
+        return response.json()
+    except:
```
Comment (NarenderRajuB) [inline apps/pre-award/lambdas/application-deadline-reminder/data.py:36]: Can we handle generic exception here HttpError.

File: apps/pre-award/lambdas/application-deadline-reminder/incomplete_application.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                round_name = round["title"]
+                round_contact_email = round.get("contact_email")
+                events = get_data_safe(
+                    Config.FUND_STORE_API_HOST
+                    + Config.FUND_EVENTS_ENDPOINT.format(
+                        fund_id=fund_id, round_id=round_id
+                    )
+                )
+
+                if events:
```
Comment (NarenderRajuB) [inline apps/pre-award/lambdas/application-deadline-reminder/incomplete_application.py:56]: can we use statement like this: 
if not events:
   continue
for event in events:
      event_type = event["type"]
      event_activation_date = datetime.strptime(
          event.get("activation_date"), "%Y-%m-%dT%H:%M:%S"
      )
      event_id = event["id"]
                        event_processed = event["processed"]

Comment (NarenderRajuB) [reply apps/pre-award/lambdas/application-deadline-reminder/incomplete_application.py:56]: which will help us in getting of nested If's

**Review by NarenderRajuB — APPROVED** [review]
LGTM but few suggestions in formatting the code to get rid of nested If's

File: apps/pre-award/lambdas/application-deadline-reminder/data.py (hunk 1 above)
Comment (ksp37-dluhc) [reply apps/pre-award/lambdas/application-deadline-reminder/data.py]: Sure, updated.

File: apps/pre-award/lambdas/application-deadline-reminder/incomplete_application.py (hunk 2 above)
Comment (ksp37-dluhc) [reply apps/pre-award/lambdas/application-deadline-reminder/incomplete_application.py]: Good idea! Updated.

**Review by Tiny49 — APPROVED** [review]
LGTM

**Pull request merged**